File: tests/primitives.rs
#![allow(
	unused_imports,
	dead_code
)]

extern crate libc;
extern crate cpython;
extern crate rustypy;

use cpython::{
	Python,
};

mod test_package;
use test_package::basics::PyModules;

use rustypy::setup_python;

#[test]
fn basics_primitives(){
	let gil = Python::acquire_gil();
	let py = gil.python();
	setup_python(py, vec!["tests", "test_package", "basics"]);
	let module: PyModules = PyModules::new(py);
	let basics = module.primitives;

	let arg = 1;
	let answ = basics.rust_bind_int_func(py, arg);
	assert_eq!(2, answ);

	let arg = 0.5;
	let answ = basics.rust_bind_float_func(py, arg);
	assert_eq!(1.0, answ);

	let arg = String::from("String from Rust, ");
	let answ = basics.rust_bind_str_func(py, arg);
	assert_eq!(
		String::from("String from Rust, added this in Python!"),
		answ);

	let arg = true;
	let answ = basics.rust_bind_bool_func(py, arg);
	assert_eq!(false, answ);

	let arg1 = String::from("String from Rust, ");
	let arg2 = 10;
	let answ = basics.rust_bind_tuple1(py, (arg1, arg2));
	assert_eq!(
		(String::from("String from Rust, added this in Python!"), 20),
		answ);

	let arg1 = String::from("String from Rust, ");
	let arg2 = true;
	let answ = basics.rust_bind_tuple2(py, (arg1, arg2));
	assert_eq!(
		(String::from("String from Rust, added this in Python!"), false),
		answ);

	let answ = basics.rust_bind_tuple3(py, 0.5, true);
	assert_eq!((1.0, false), answ);
}

File: src/rslib/tests/test_package/basics/primitives.py
import typing as typ

# primitives:


def rust_bind_int_func(int_arg: int) -> int:
    rval = int_arg + 1
    return rval


def rust_bind_float_func(float_arg: float) -> float:
    rval = float_arg + 0.5
    return rval


def rust_bind_str_func(str_arg: str) -> str:
    rval = str_arg + "added this in Python!"
    return rval


def rust_bind_bool_func(bool_arg: bool) -> bool:
    rval = False
    return rval


def rust_bind_tuple1(tup_arg: typ.Tuple[str, int]) -> typ.Tuple[str, int]:
    output = list(tup_arg)
    output[0] += "added this in Python!"
    output[1] += 10
    output = tuple(output)
    return output

K = typ.Tuple[str, bool]


def rust_bind_tuple2(tup_arg: K) -> K:
    output = list(tup_arg)
    output[0] += "added this in Python!"
    if not output[1]:
        raise AssertionError(output[1])
    else:
        output[1] = False
    output = tuple(output)
    return output

J = typ.Tuple[float, bool]


def rust_bind_tuple3(tup_arg1: float, tup_arg2: bool) -> J:
    out_arg1 = tup_arg1 + 0.5
    if not tup_arg2:
        raise AssertionError(tup_arg2)
    else:
        out_arg2 = False
    return out_arg1, out_arg2
